mbedtls/src/pk/mod.rs

```diff
@@ -236,7 +185,7 @@ impl Pk {
     /// Takes both DER and PEM forms of PKCS#1 or PKCS#8 encoded keys.
     ///
     /// When calling on PEM-encoded data, `key` must be NULL-terminated
-    pub fn from_private_key(key: &[u8], password: Option<&[u8]>) -> Result<Pk> {
+    pub fn from_private_key<F: Random>(rng: &mut F, key: &[u8], password: Option<&[u8]>) -> Result<Pk> {
```

Comment:
This one is odd from the user perspective. I have some key bytes and in order to read them, I need to provide an RNG.

Randomness here is actually needed for computing the public key given the read private key bytes (for instance, in EC, private key is a scalar `d` and public key is `dP`, the scalar multiplication of  `d`. To protect it against side channels, blinding is used when multiplying, which requires randomness).

Ideally, IMHO, this function should only read the private bytes (no randomness), and compute/cache the public key (with the RNG) when a function is called to compute the public key for a given `Pk` (e.g. `.public_key`. I don't know if this is currently possible, given MbedTLS's plans about a global RNG. We should check the internals of `pk_parse_key` and see if there are similar funcions that only read the private key.